week1/wordcount_done.py

```diff
@@ -0,0 +1,48 @@
+import sys
+def get_dictionary(filename):
+    file = open(filename, "r")
```

Comment:
It is good practice to use the with keyword when dealing with file objects. The advantage is that the file is properly closed after its suite finishes, even if an exception is raised at some point. Using with is also much shorter than writing equivalent try-finally blocks [[1]](https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files):
```python
with open('workfile') as f:
    read_data = f.read()
```